cpu/stm32/cpu_common.c

```diff
@@ -51,343 +75,350 @@ static const uint8_t apbmul[] = {
+    #define RCC_REG_IS_ATOMIC 1
+    case APB12:
+        return &APB12_PERIPH_DIS;
+#endif
+#ifdef APB2_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case APB2:
+        return &APB2_PERIPH_DIS;
+#endif
+#ifdef APB22_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case APB22:
+        return &APB22_PERIPH_DIS;
+#endif
+#ifdef APB3_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case APB3:
+        return &APB3_PERIPH_DIS;
+#endif
+#ifdef APB4_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case APB4:
+        return &APB4_PERIPH_DIS;
+#endif
+#ifdef AHB_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB:
+        return &AHB_PERIPH_DIS;
+#endif
+#ifdef AHB1_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB1:
+        return &AHB1_PERIPH_DIS;
+#endif
+#ifdef AHB2_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB2:
+        return &AHB2_PERIPH_DIS;
+#endif
+#ifdef AHB22_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB22:
+        return &AHB22_PERIPH_DIS;
+#endif
+#ifdef AHB3_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB3:
+        return &AHB3_PERIPH_DIS;
+#endif
+#ifdef AHB4_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB4:
+        return &AHB4_PERIPH_DIS;
+#endif
+#ifdef IOP_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case IOP:
+        return &IOP_PERIPH_DIS;
 #endif
-#if defined(CPU_FAM_STM32WL) || defined(CPU_FAM_STM32U5)
-        case APB3:
-            RCC->APB3ENR &= ~(mask);
-            break;
-#endif
-#if defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32WB) || \
-    defined(CPU_FAM_STM32G4) || defined(CPU_FAM_STM32L5) || \
-    defined(CPU_FAM_STM32U5) || defined(CPU_FAM_STM32WL)
-        case APB12:
-            RCC->APB1ENR2 &= ~(mask);
-            break;
-#elif defined(CPU_FAM_STM32G0) || defined(CPU_FAM_STM32C0)
-        case APB12:
-            RCC->APBENR2 &= ~(mask);
-            break;
-#endif
-#if defined(CPU_FAM_STM32L0)
-        case AHB:
-            RCC->AHBENR &= ~(mask);
-            break;
-        case IOP:
-            RCC->IOPENR &= ~(mask);
-            break;
-#elif defined(CPU_FAM_STM32L1) || defined(CPU_FAM_STM32F1) || \
-      defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F3)
-        case AHB:
-            RCC->AHBENR &= ~(mask);
-            break;
-#elif defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
-      defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32F7) || \
-      defined(CPU_FAM_STM32WB) || defined(CPU_FAM_STM32G4) || \
-      defined(CPU_FAM_STM32L5) || defined(CPU_FAM_STM32U5) || \
-      defined(CPU_FAM_STM32WL)
-        case AHB1:
-            RCC->AHB1ENR &= ~(mask);
-            break;
-/* STM32F410 RCC doesn't provide AHB2 and AHB3 */
-#if !defined(CPU_LINE_STM32F410Rx)
-        case AHB2:
-#if defined(CPU_FAM_STM32U5)
-            RCC->AHB2ENR1 &= ~(mask);
+
+    /* If MCU does not have separate set/clear bits. */
+#if RCC_REG_IS_ATOMIC == 0
+    default:
+        return  _rcc_dis_reg(bus);
```

Comment:
```suggestion
        return _rcc_dis_reg(bus);
```

Reply:
Fixed. Also, the call should have been to `_rcc_en_reg()`, which is also now fixed.